<launch>
    <arg name="debug"               default="false"/>
    <arg name="frame" default="ydlidar_link"/>
    <arg name="bag_filename"        default="$(env HOME)/bags/autolabor_bag/20200507_elevator_position_dynamic_1.bag"/>
    <!-- /////////////////////////////////////////////////////////////////////////////////// -->
    <group if="$(arg debug)">

        <param name="use_sim_time" value="true" />
        <node name="playbag" pkg="rosbag" type="play"
                args="--clock $(arg bag_filename)" />

        <param name="robot_description"
            textfile="$(find autolabor_description)/urdf/autolabor.urdf" />

        <node name="robot_state_publisher" pkg="robot_state_publisher"
            type="robot_state_publisher" />


        <!-- rviz -->
        <node name="rviz" pkg="rviz" type="rviz" args="-d $(find autolabor_move)/rviz/elevator.rviz" required="true"/>

    </group>

    <!-- <node pkg="nodelet" type="nodelet" name="pcl_manager" args="manager" output="screen" />
    <node pkg="nodelet" type="nodelet" name="voxel_grid" args="load pcl/VoxelGrid pcl_manager" output="screen">
        <remap from="~input"  to="velodyne_points" />
        <remap from="~output" to="filtered_points" />
        <rosparam>
        filter_field_name: z
        filter_limit_min: 0
        filter_limit_max: 0.5
        filter_limit_negative: False
        </rosparam>
    </node> -->

    <node pkg="autolabor_move" type="elevator_position_finder" name="elevator_position_finder" output="screen">
        <remap from="frame"    to="$(arg frame)" />
        <param name="window_type" value="circular" />
        <param name="window_size" value="0.15" />
        <param name="max_scan_range" value="6.0" />
        <param name="corner_feature_threshold" value="0.0000025" />
        <param name="elevator_feature_dis_range" value="0.2" />
        <param name="elevator_feature_d_angle_range" value="0.4" />
        
    </node>

    <node pkg="autolabor_move" type="elevator_pose_tf" name="elevator_pose_tf" output="screen">
        <param name="source_frame_id" value="odom" />
        <param name="target_frame_id" value="base_link" />
        <param name="pub_tf" value="true" />
        <param name="pub_odom" value="false" />
        
    </node>

</launch>
